packages/medusa-file-local/src/services/__tests__/local-file-service.spec.ts

```diff
@@ -0,0 +1,140 @@
+import fs from "fs"
+import fsp from "fs/promises"
+import stream from "stream"
+
+jest.mock("fs")
+jest.mock("fs/promises")
+
+const mockedFs = fs as jest.Mocked<typeof fs>
+
+import LocalFileService from "../local-file-service"
+
+const defaultBackendUrl = "http://localhost:9000"
+const defaultUploadDir = "uploads/images"
+
+describe("LocalFileService", () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("getUploadStreamDescriptor", () => {
+    it("should return an object with a writeStream and a promise", async () => {
+      const fileData = {
+        name: "test",
+        ext: "png",
+      }
+
+      mockedFs.createWriteStream.mockReturnValue(
+        new stream.PassThrough() as any
+      )
+
+      const service = new LocalFileService({})
+      const result = await service.getUploadStreamDescriptor(fileData)
+
+      expect(fsp.mkdir).toHaveBeenCalledWith(defaultUploadDir, {
+        recursive: true,
+      })
+      expect(fs.createWriteStream).toHaveBeenCalledWith(
+        `${defaultUploadDir}/${fileData.name}.${fileData.ext}`
+      )
+
+      expect(result).toEqual({
+        writeStream: expect.any(stream.PassThrough),
+        promise: expect.objectContaining({
+          then: expect.any(Function),
+        }),
```

Comment:
Due to the well-known issue of [Jest](https://github.com/jestjs/jest/issues/2549) I couldn't simply use `expect.any(Promise)`, so I had to use some workaround.